### PR TITLE
Geomap: styleConfig cleanup and symbol caching

### PR DESCRIPTION
**Pull request by ryantxu**
This PR cleans up how geomap creates styles for each feature.

This moves everything to a "style" folder and adds a getter for each symbol type.

This started as minor clean up to #41565 -- but I quickly realized that we really want the *same* style object/config for: markers, text, and geojson.  In fact since openlayers is able to show text+symbol in one `Style` object, we should really combine things so it is easy to do that

### CODE REVIEW

File: public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts [hunk 1]
```diff
@@ -77,6 +78,11 @@ export const geojsonMapper: MapLayerRegistryItem<GeoJSONMapperConfig> = {
     const vectorLayer = new VectorLayer({
       source,
       style: (feature: FeatureLike) => {
+        const type = feature.getGeometry()?.getType();
+        if (type === 'Point') {
+          return circleMarker({color:DEFAULT_STYLE_RULE.fillColor});
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts:83]: current behavior is that nothing at all is shown for point data (load the airports to test)

File: public/app/plugins/panel/geomap/style/markers.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                  points: 4,
+                  radius: cfg.size,
+                  angle: Math.PI / 4,
+                }),
+              }),
+            ];
+          }
+        : errorMarker,
+    };
+    markerMakers.register(maker);
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/style/markers.ts:251]: each URL will only try to load once :taco:

File: public/app/plugins/panel/geomap/migrations.ts [hunk 3]
```diff
@@ -100,23 +104,38 @@ function asNumber(v: any): number | undefined {
 }
 
 export const mapMigrationHandler = (panel: PanelModel): Partial<GeomapPanelOptions> => {
-  const pluginVersion = panel?.pluginVersion;
-  if (pluginVersion?.startsWith('8.1') || pluginVersion?.startsWith('8.2') || pluginVersion?.startsWith('8.3')) {
-    if (panel.options?.layers?.length > 0) {
+  const pluginVersion = panel?.pluginVersion ?? '';
+
+  // before 8.3, only one layer was supported!
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/migrations.ts:109]: Do we have any external places for documenting versioning of panels / plugins and their related constraints? I could see this becoming challenging to manage at more scale

File: public/app/plugins/panel/geomap/style/markers.ts (hunk 2 above)
Comment (nmarrs) [reply public/app/plugins/panel/geomap/style/markers.ts:251]: Were URLs loading every time we were rendering before?

**Review by nmarrs — COMMENTED** [review]
It may make sense to walk over these changes over a quick huddle so I can better understand the context (and more confidently approve ;)

**Review by nmarrs — APPROVED** [review]

**Pull request merged**